tensorflow_model_optimization/python/core/quantization/keras/collaborative_optimizations/cluster_preserve/cluster_preserve_quantize_registry_test.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for ClusterPreserveQuantizeRegistry."""
+
+from absl.testing import parameterized
+
+import tensorflow as tf
+
+from tensorflow.python.keras import keras_parameterized
+
+from tensorflow_model_optimization.python.core.quantization.keras import quantize_config
+from tensorflow_model_optimization.python.core.clustering.keras import clustering_registry
+from tensorflow_model_optimization.python.core.quantization.keras.default_8bit import default_8bit_quantize_registry
+from tensorflow_model_optimization.python.core.quantization.keras.collaborative_optimizations.cluster_preserve import cluster_preserve_quantize_registry
+
+QuantizeConfig = quantize_config.QuantizeConfig
+layers = tf.keras.layers
+
+
+@keras_parameterized.run_all_keras_modes
+class ClusterPreserveQuantizeRegistryTest(tf.test.TestCase,
+                                        parameterized.TestCase):
+  def setUp(self):
+    super(ClusterPreserveQuantizeRegistryTest, self).setUp()
+    self.cluster_preserve_quantize_registry = \
+      cluster_preserve_quantize_registry.ClusterPreserveQuantizeRegistry()
+    # layers which are supported
+    # initial and build a Conv2D layer
+    self.layer_conv2d = layers.Conv2D(10, (2, 2))
+    self.layer_conv2d.build((2, 2))
+    # initial and build a Dense layer
+    self.layer_dense = layers.Dense(10)
+    self.layer_dense.build((2, 2))
+    # initial and build a ReLU layer
+    self.layer_relu = layers.ReLU()
+    self.layer_relu.build((2, 2))
+
+    # a layer which is not supported
+    # initial and build a Custom layer
+    self.layer_custom = self.CustomLayer()
+    self.layer_custom.build()
+
+
+  class CustomLayer(layers.Layer):
+    # a simple custom layer with training weights
+    def __init__(self):
+      super(ClusterPreserveQuantizeRegistryTest.CustomLayer, self).__init__()
+
+    def build(self, input_shape=(2, 2)):
+      self.add_weight(shape=input_shape,
+                      initializer="random_normal",
+                      trainable=True)
+
```

Comment:
Should be a double-space after classes

Reply:
For this one, I am not sure. Does the inner class (not a top-level definition) follow the same rule of 2 blank lines? I checked the coding style given to us before, and it says:

> Two blank lines between top-level definitions, be they function or class definitions. One blank line between method definitions and between the class line and the first method. No blank line following a def line. Use single blank lines as you judge appropriate within functions or methods.

Is there any other coding style you are using internally atm? Also, I used the pylint [here ](https://www.tensorflow.org/community/contribute/code_style) and it never reported issues of lines or spaces. Therefore, I switched to `pycodestyle`. Is there a tool integrated with your current coding style for TFMOT that we could use in the future?